src/main/java/core/basesyntax/service/TransactionParser.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.Transaction;
+
+public class TransactionParser {
+    public Transaction parseLine(String line) {
+        String[] parts = line.split(",");
+        String operation = parts[0];
+        String fruitName = parts[1];
+        int quantity = Integer.parseInt(parts[2]);
+        return new Transaction(operation, fruitName, quantity);
```

Comment:
Consider adding validation for the input `line` such as checking if it's not null or empty, and whether it contains the expected number of parts after splitting. This will help to prevent `ArrayIndexOutOfBoundsException` if the input format is not as expected.